components/layout/construct.rs

```diff
@@ -222,7 +248,7 @@ impl InlineFragmentsAccumulator {
                 pseudo: node.get_pseudo_element_type().strip(),
                 style: node.style(style_context),
                 selected_style: node.selected_style(),
-                flags: InlineFragmentNodeFlags::empty(),
+                flags: FIRST_FRAGMENT_OF_ELEMENT | LAST_FRAGMENT_OF_ELEMENT,
```

Comment:
This looks good to me, but I wonder if there's a way to organize this so this doesn't start with this flags. It feels weird to use these flags here since we always remove them in `to_intermediate_inline_fragments`.

Maybe keeping in the accumulator something like `for_ib_split` flag or similar, and don't insert the flags in that case?